src/NServiceBus.Persistence.CosmosDB/Outbox/OutboxPersister.cs

```diff
@@ -39,7 +39,7 @@ public async Task<OutboxMessage> Get(string messageId, ContextBag context, Cance
                 return null;
             }
 
-            partitionKey = new PartitionKey(messageId);
+            partitionKey = new PartitionKey(endpointName);
```

Comment:
@danielmarbach I've done the change we talked about in this PR since we have to add code to adjust the endpoint name based on tx-session configuration.

Reply:
@cquirosj @poornimanayar FYI: This line will only be reached when using callbacks, so therefore this change should be safe since callbacks are volatile by nature so changing the partition key affecting inflight messages does not need any migration guidance

Reply:
Should this be a combination of message id and endpoint name?